aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -82,14 +88,33 @@ void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
         if (nv.isNone()) {
           continue;
         }
-        auto* interpreter = nv.unsafeToTensorImpl()->pyobj_slot()->pyobj_interpreter();
-        if (interpreter) {
-          (*interpreter)->dispatch(op, stack);
-          return;
+
+        auto* t = nv.unsafeToTensorImpl();
+        if (t->key_set().has(c10::DispatchKey::Python)) {
+          tensors_with_python_key_present = true;
+        }
+
+        auto* t_interpreter = t->pyobj_slot()->pyobj_interpreter();
+        if (!interpreter && t_interpreter) {
```

Comment:
nit: right now you are calling `t->pyobj_slot()->pyobj_interpreter()` on every single tensor arg, but you can fastpath and only grab it in the case when `interpreter` isn't already set.

(I know some folks are worried about torch_dispatch related cpu overhead - so I'm not sure how much this matters but anything we can do to avoid unnecessary work in this codepath seems probably worth doing

Reply:
agree, I didn't originally realize it was expensive to query for the interpreter, but it makes sense. moved :)